.github/actions/elixir_cache/action.yml

```diff
@@ -1,19 +1,44 @@
 name: Elixir Cache
 description: Elixir Cache
 
+on:
+  push:
+    branches:
+      - develop
+      #cron data to be provided
+  schedule:
+    - cron: '0 0 * * *'
+
 inputs:
   job_name:
     description: Extra key to use for restoring and saving the cache
     required: true
 
-runs:
-  using: composite
-  steps:
-    - uses: actions/cache@67b839edb68371cc5014f6cea11c9aa77238de78
-      with:
-        path: |
-          **/deps
-          **/_build
-        key: cache-elixir-${{ github.workflow }}-${{ github.job }}-${{ hashFiles('**/mix.lock') }}
-        restore-keys: |
-          cache-elixir-${{ github.workflow }}-${{ github.job }}-
+jobs:
+  cache:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if triggered by schedule
+        run: echo "::set-output name=schedule::${{ github.event_name == 'schedule' }}"
+        id: schedule
```

Comment:
I don't think there's a need for this step, we can call all other steps conditionally as `${{ github.event_name == 'schedule' }}`, please check below review.